File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int i,j=0,k=0;
       sort(nums.begin(),nums.end());
       for(i=0;i<nums.size()-1;i++)
       {
        if(nums[i+1]-nums[i]==0)
        {
          continue;
        }
        else if(nums[i+1]-nums[i]==1)
       j++;
        else{
            if(j>k)
            k=j;
            j=0;
        }
        
       }
       if(j>k)
       k=j;
       return (k+1);
    
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # Create a set for O(1) lookups
        num_set = set(nums)
        longest_streak = 0

        for num in num_set:
            # Check if 'num' is the start of a sequence
            # If num - 1 is in the set, 'num' is NOT the start
            if (num - 1) not in num_set:
                current_num = num
                current_streak = 1

                # Keep checking for the next numbers in the sequence
                while (current_num + 1) in num_set:
                    current_num += 1
                    current_streak += 1

                # Update our global maximum length
                longest_streak = max(longest_streak, current_streak)
        
        return longest_streak